### ISSUE TITLE
Output eval prediction

### ISSUE THREAD

**Issue by pltrdy** [tensorflow/tensor2tensor#958]
Hi,

Is there a way (more or less "hacky") to output predictions that are made during the evaluation step?
I would find it useful to monitor how the model is performing, qualitatively.

**Comment by martinpopel** [comment]
yes, turn off the internal evaluation (`--schedule=train`) and use `t2t-translate-all` instead (plus `t2t-bleu` for seeing the *real* BLEU in tensorboard)

**Comment by pltrdy** [comment]
Thanks Martin, I was thinking about an option at training time, to monitor performance on validation (not test) set. Since I'm not keeping each checkpoint it may not be trivial to mimic this a posteriori.

**Comment by pltrdy** [comment]
Thanks, interesting feature.

I'm not sure how to translate from the validation (dev) set, since the split train/dev is done by t2t. I do have some kind of encoded 'dev' files in my data directory, is there an easy way to generate text files from these (or directly translate from t2t-generated files?)

**Comment by martinpopel** [comment]
The train/dev split by T2T exactly the way how you specified it in Problem definition.
However, with `t2t-translate-all` you can translate whatever file you want (plain-text, one sentence per line).
Note that `t2t-translate-all` can run "on the background" continuously during the whole training and monitor the model directory for new checkpoints.
So you can set up T2T to keep just N last checkpoints (`t2t-translate-all` can run on CPU - e.g. on a different machine than the main training).
Similarly, `t2t-bleu` and `t2t-avg-all` support monitoring a directory for new files.

**Comment by xuekun90** [comment]
@martinpopel , I met some issue, when the training is still in progress, I run the t2t-avg-all, it will OOM. It seems the training has used 95% memory of the GPU, so there is no enough memory for t2t-avg-all?
So the t2t-avg-all could not use during the training?

**Comment by martinpopel** [comment]
@xuekun90: I always run `t2t-avg-all` on different machine than the main training. Moreover, for `t2t-avg-all` there is no speedup when running on GPU, so I use a CPU-only machine.
If you have just one machine, you can try running `t2t-avg-all` with CUDA_VISIBLE_DEVICES set to empty string (or under a Python virtual env which has tensorflow instead of tensorflow-gpu).
This way `t2t-avg-all` will use CPU memory only (and the training can use all the available GPU memory).

**Comment by xuekun90** [comment]
@martinpopel: thanks,  i will have a try.
